tests/test_widgets.py

```diff
@@ -2184,6 +2183,124 @@ def testNoneSamplePropertiesDialog(self):
+                'parameter_count': 3,
+                'mesh_args_default': {
+                    'width': 50.000,
+                    'height': 100.000,
+                    'depth': 200.000
+                },
+                'mesh_args_custom': {
+                    'width': 900.000,
+                    'height': 330.000,
+                    'depth': 700.000
+                }
+            },
+            {
+                'case': 'Cylinder',
+                'primitive': Primitives.Cylinder,
+                'parameter_count': 2,
+                'mesh_args_default': {
+                    'radius': 100.000,
+                    'height': 200.000
+                },
+                'mesh_args_custom': {
+                    'radius': 450.000,
+                    'height': 50.000
+                }
+            },
+            {
+                'case': 'Sphere',
+                'primitive': Primitives.Sphere,
+                'parameter_count': 1,
+                'mesh_args_default': {
+                    'radius': 100.000
+                },
+                'mesh_args_custom': {
+                    'radius': 600.000
+                }
+            },
+            {
+                'case': 'Tube',
+                'primitive': Primitives.Tube,
+                'parameter_count': 3,
+                'mesh_args_default': {
+                    'outer_radius': 100.000,
+                    'inner_radius': 50.000,
+                    'height': 200.000
+                },
+                'mesh_args_custom': {
+                    'outer_radius': 300.000,
+                    'inner_radius': 150.000,
+                    'height': 1000.000
+                }
+            },
+        )
+
+        for test_case in primitive_test_cases:
+            with self.subTest(test_case['case']):
+                self.dialog = InsertPrimitiveDialog(test_case['primitive'], self.view)
+                self.assertTrue(self.dialog.create_primitive_button.isEnabled())
+
+                # Testing primitive creation with default values
+                self.dialog.create_primitive_button.click()
+                self.view.presenter.addPrimitive.assert_called_with(test_case['primitive'],
+                                                                    test_case['mesh_args_default'])
+
+                # Testing primitive creation with custom values
+                for parameter, value in test_case['mesh_args_custom'].items():
+                    self.dialog.textboxes[parameter].value = value
+                self.dialog.create_primitive_button.click()
+                self.view.presenter.addPrimitive.assert_called_with(test_case['primitive'],
+                                                                    test_case['mesh_args_custom'])
+
+                # Testing with zero and negative values
+                for ix in range(test_case['parameter_count']):
+                    control = self.dialog.form_group.form_controls[ix]
```

Comment:
This test will always be correct if the first control (ix=0) is invalid and will never fail even if the second control is valid (ix=1)